include/z64audio.h

```diff
@@ -552,32 +565,42 @@ typedef struct {
 } ReverbSettings; // size = 0x18
 
+/**
+ * The high-level audio specifications requested when initializing or resetting the audio heap.
+ * Most often resets during scene transitions, but will highly depend on game play.
+ */ 
 typedef struct {
-    /* 0x00 */ u32 frequency;
+    /* 0x00 */ u32 frequency; // Target sampling rate in Hz
     /* 0x04 */ u8 unk_04;
     /* 0x05 */ u8 numNotes;
     /* 0x06 */ u8 numSequencePlayers;
     /* 0x07 */ u8 unk_07; // unused, set to zero
     /* 0x08 */ u8 unk_08; // unused, set to zero
     /* 0x09 */ u8 numReverbs;
     /* 0x0C */ ReverbSettings* reverbSettings;
-    /* 0x10 */ u16 sampleDmaBufSize1;
-    /* 0x12 */ u16 sampleDmaBufSize2;
+    /* 0x10 */ u16 sampleDmaBufSize1; // size of buffer in the audio misc pool to store small snippets of indivisual samples. Stored short-lived.
+    /* 0x12 */ u16 sampleDmaBufSize2; // size of buffer in the audio misc pool to store small snippets of indivisual samples. Stored long-lived.
     /* 0x14 */ u16 unk_14;
-    /* 0x18 */ u32 persistentSeqMem;
-    /* 0x1C */ u32 persistentFontMem;
-    /* 0x20 */ u32 persistentSampleMem;
-    /* 0x24 */ u32 temporarySeqMem;
-    /* 0x28 */ u32 temporaryFontMem;
-    /* 0x2C */ u32 temporarySampleMem;
-    /* 0x30 */ s32 persistentSampleCacheMem;
-    /* 0x34 */ s32 temporarySampleCacheMem;
+    /* 0x18 */ u32 persistentSeqCacheSize;  // size of cache on audio pool to store sequences persistently
+    /* 0x1C */ u32 persistentFontCacheSize; // size of cache on audio pool to store soundFonts persistently
+    /* 0x20 */ u32 persistentSampleBankCacheSize; // size of cache on audio pool to store entire sample banks persistently
+    /* 0x24 */ u32 temporarySeqCacheSize;  // size of cache on audio pool to store sequences temporarily
+    /* 0x28 */ u32 temporaryFontCacheSize; // size of cache on audio pool to store soundFonts temporarily
+    /* 0x2C */ u32 temporarySampleBankCacheSize; // size of cache on audio pool to store entire sample banks temporarily
+    /* 0x30 */ s32 persistentSampleCacheSize; // size of cache in the audio misc pool to store individual samples persistently
+    /* 0x34 */ s32 temporarySampleCacheSize; // size of cache in the audio misc pool to store individual samples temporarily
 } AudioSpec; // size = 0x38
 
+/**
+ * The audio buffer stores the fully processed digital audio before it is sent to the audio interface (AI), then to the
+ * digital-analog converter (DAC), then to play on the speakers. The audio buffer is written to by the rsp after
+ * processing audio commands, and the audio buffer is read by AI which sends the data to the DAC.
+ * This struct parameterizes that buffer.
+ */
 typedef struct {
     /* 0x00 */ s16 specUnk4;
-    /* 0x02 */ u16 frequency;
-    /* 0x04 */ u16 aiFrequency;
+    /* 0x02 */ u16 frequency; // Target sampling rate in Hz
+    /* 0x04 */ u16 aiFrequency; // True sampling rate set to the audio interface (AI) for the audio digital-analog converter (DAC)
```

Comment:
"set to the audio interface" ? does that mean "sent to" or "used by"?

Reply:
`aiFrequency = osAiSetFrequency(frequency)`

And we have ultra64 docs on that:

"Based on the input requested frequency (in Hz), the osAiSetFrequency function calculates the correct values for the internal divisors and returns the actual frequency generated by these divisors. The requested frequency should fall between 3000 Hz and 368000 Hz on NTSC systems, or between 3050 Hz and 376000 Hz on PAL systems."

http://ultra64.ca/files/documentation/online-manuals/functions_reference_manual_2.0i/os/osAiSetFrequency.html

Reply:
I think "set to" would be better as "used by" then

"set to" sounds like you would set something with it, and also "set to" sounds weird in the phrasing (usually "A set to B" means `A=B`)

Reply:
id think "set on" would be best, probably
the audio interface is an actual chip on the n64 right, and these functions are setting properties on/for it

Reply:
I believe the audio interface is a chip on the coprocessor yeah (can't remember if it's the rsp or rdp).

aiFrequency is more of an output of `osAiSetFrequency` instead of being used to set something. The values that are sent to the audio interface are:
```
HW_REG(AI_DACRATE_REG, u32) = dacRate - 1;
HW_REG(AI_BITRATE_REG, u32) = bitrate - 1;
```
inside `osAiSetFrequency`, and it just returns the frequency for use elsewhere:
`return osViClock / (s32)dacRate;`

I don't 100% follow `osAiSetFrequency` but `aiFrequency` is really only used once in all of the audio library, which eventually leads to timing in sequences, which isn't directly related to the audio interface, it's more of an output from the audio interface

Reply:
after a second read it should probably be "true sampling rate of the Audio Interface"?
`frequency` is used to calculate `dacRate` and `bitRate`, which is set on the audio interface, and the return is what the resulting frequency is using those two divisors

Reply:
Also for reference, I've called these `samplingFrequency` in MM since this is actually the frequency in Hz as opposed to the frequency commonly used in Audio which is relative to `1.0f`
```
u16 samplingFreq;
u16 aiSamplingFreq;
```
I didn't rename as this PR was already getting too large and was going to save it for later. Since we're discussing these variables now though, I could always propose to change it now

Reply:
fine by me